### PR TITLE
Data1193 fix type error - Add connection id to show tables.

### PR DESCRIPTION
**Pull request by udaybhanu28**
Added connection id to show tables.

### CODE REVIEW

**Comment by udaybhanu28** [comment]
Updated error handling to be more generic rather than just for show tables, The changes in this PR would log DEBUG level logs on **any query** failure. Below is a log of a test that I did on staging by intentionally updating the SHOW TABLES command.

```
2020-06-12T10:12:15.946Z - error: Error in : SHOW TABLS
2020-06-12T10:12:15.946Z - error: Error message : SQL compilation error:
syntax error line 1 at position 5 unexpected 'TABLS'.
2020-06-12T10:12:15.946Z - debug: Query Id : 0194db04-03ee-552e-0000-291d1b0cec5a Payload Info : null
2020-06-12T10:12:15.946Z - debug: Connection Id : 3aed235a-059e-4d26-86b6-a25ec82e1d5f

File: lib/connections/snowflake.js [hunk 1]
```diff
@@ -157,11 +159,15 @@ connection.prototype.query = function(query, data, callback, skipTransactions){
                     if(String(err.message).includes('Empty SQL statement')) {
                         callback(null);
                     } else {
+                        self.book.logger.log('Error in : ' + String(query), 'error');
+                        self.book.logger.log('Error message : ' + err.message, 'error');
+                        self.book.logger.log('Query Id : ' + stmt.getStatementId() + ' Payload Info : ' + String(data), 'debug');
+                        self.book.logger.log('Connection Id : ' + self.connection_id, 'debug');
```
Comment (yogeshbhate) [inline lib/connections/snowflake.js]: this should also be error level log

File: lib/connections/snowflake.js [hunk 2]
```diff
@@ -157,11 +159,15 @@ connection.prototype.query = function(query, data, callback, skipTransactions){
                     if(String(err.message).includes('Empty SQL statement')) {
                         callback(null);
                     } else {
+                        self.book.logger.log('Error in : ' + String(query), 'error');
+                        self.book.logger.log('Error message : ' + err.message, 'error');
+                        self.book.logger.log('Query Id : ' + stmt.getStatementId() + ' Payload Info : ' + String(data), 'debug');
```
Comment (yogeshbhate) [inline lib/connections/snowflake.js]: this should also be error level log

File: lib/connections/snowflake.js [hunk 3]
```diff
@@ -36,8 +37,10 @@ connection.prototype.connect = function(callback){
     self.connection = snowflake.createConnection(self.options);
 
     self.connection.connect(function(err, conn) {
+        self.connection_id = conn.getId();
         if (err) {
             self.book.logger.log('Unable to connect: ' + err.message, 'error');
+            self.book.logger.log('Connection id :  ' + self.connection_id, 'debug');
```
Comment (yogeshbhate) [inline lib/connections/snowflake.js]: this should also be error level log

File: lib/connections/snowflake.js [hunk 4]
```diff
@@ -36,8 +37,10 @@ connection.prototype.connect = function(callback){
     self.connection = snowflake.createConnection(self.options);
 
     self.connection.connect(function(err, conn) {
+        self.connection_id = conn.getId();
```
Comment (yogeshbhate) [inline lib/connections/snowflake.js]: we should first check for err, if not error then we should do this.. i think its a wrong place to put this line. Also log the connection id as info when the connection is successful.

**Review by yogeshbhate — CHANGES_REQUESTED** [review]

**Review by rajpalparyani — APPROVED** [review]
LGTM. As review comments are also incorporated,

**Pull request merged**